lua/luasnip/nodes/snippet.lua

```diff
@@ -689,12 +689,7 @@ function Snippet:fake_expand(opts)
 	if opts.env then
 		self.env = opts.env
 	else
-		self.env = {}
-		setmetatable(self.env, {
-			__index = function(_, key)
-				return Environ.is_table(key) and { "$" .. key } or "$" .. key
-			end,
-		})
+		self.env = Environ.fake()
```

Comment:
Oh right, also good that you moved this into `environ.lua`, much better than creating it in such a disconnected place.